src/main/java/com/academia/library/service/impl/TagServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package com.academia.library.service.impl;
+
+import com.academia.library.dto.TagRequestDto;
+import com.academia.library.dto.TagResponseDto;
+import com.academia.library.exception.TagAlreadyExistException;
+import com.academia.library.exception.TagNotFoundException;
+import com.academia.library.mapper.TagMapper;
+import com.academia.library.model.Tag;
+import com.academia.library.repository.TagRepository;
+import com.academia.library.service.TagService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
```

Comment:
Follow comments in other service classes to refactor this class

Reply:
Ok, thank